providers/os/resources/npm.go

```diff
@@ -0,0 +1,448 @@
+
+				if !f.IsDir() {
+					continue
+				}
+
+				// check if there is a package-lock.json or package.json file
+				packageLockPath := filepath.Join(nodeModulesPath, p, "/package-lock.json")
+				packageJsonPath := filepath.Join(nodeModulesPath, p, "/package.json")
+
+				packageLockExists, _ := afs.Exists(packageLockPath)
+				packageJsonExists, _ := afs.Exists(packageJsonPath)
+
+				// add files to evidence
+				if packageLockExists {
+					evidenceFiles = append(evidenceFiles, packageLockPath)
+				}
+				if packageJsonExists {
+					evidenceFiles = append(evidenceFiles, packageJsonPath)
+				}
+
+				// parse npm files
+				if packageLockExists {
+					log.Debug().Str("path", packageLockPath).Msg("found package-lock.json file")
+					f, err := getFileContent(r.MqlRuntime, packageLockPath)
+					if err != nil {
+						continue
+					}
+					content := f.GetContent()
+					if content.Error != nil {
+						continue
+					}
+
+					p := &npm.PackageLockParser{}
+					info, err := p.Parse(strings.NewReader(content.Data), packageLockPath)
+					if err != nil {
+						log.Error().Err(err).Str("path", packageLockPath).Msg("could not parse package-lock.json file")
+					}
+					root := info.Root()
+					if root != nil {
+						directPackageList = append(directPackageList, root)
+					}
+					transitive := info.Transitive()
+					if transitive != nil {
+						transitivePackageList = append(transitivePackageList, transitive...)
+					}
+
+				} else if packageJsonExists {
+					log.Debug().Str("path", packageJsonPath).Msg("found package.json file")
+					f, err := getFileContent(r.MqlRuntime, packageJsonPath)
+					if err != nil {
+						continue
+					}
+					content := f.GetContent()
+					if content.Error != nil {
+						continue
+					}
+
+					p := &npm.PackageJsonParser{}
+					info, err := p.Parse(strings.NewReader(content.Data), packageJsonPath)
+					if err != nil {
+						log.Error().Err(err).Str("path", packageJsonPath).Msg("could not parse package.json file")
+					}
+					root := info.Root()
+					if root != nil {
+						directPackageList = append(directPackageList, root)
+					}
+					transitive := info.Transitive()
+					if transitive != nil {
+						transitivePackageList = append(transitivePackageList, transitive...)
+					}
+				}
+			}
+		}
+	}
+	return directPackageList, transitivePackageList, evidenceFiles, nil
+}
+
+func (r *mqlNpmPackages) gatherPackagesFromLocation(conn shared.Connection, path string) (*npm.Package, []*npm.Package, []*npm.Package, []string, error) {
+	evidenceFiles := []string{}
+
+	// specific path was provided
+	afs := &afero.Afero{Fs: conn.FileSystem()}
+	isDir, err := afs.IsDir(path)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	loadPackageLock := false
+	packageLockPath := ""
+	loadPackageJson := false
+	packageJsonPath := ""
+
+	if isDir {
+		// check if there is a package-lock.json or package.json file
+		packageLockPath = filepath.Join(path, "/package-lock.json")
```

Comment:
[nit] i don't think you need the forward slash

Reply:
will remove those